autosklearn/metrics/__init__.py

```diff
@@ -388,14 +394,20 @@ def calculate_score(
                         continue
                     else:
                         raise e
+
+        if metric and metric.name not in score_dict.keys():
```

Comment:
Together with the proposed change above, this should become redundant, right?

Reply:
This change is added to make sure metric passed is calculated. It may possible that the list of scorers does not contain metric scorer required for train/test scoring.

Reply:
Got it, thanks a lot. Another small question, can `metric` really be `None`?

Reply:
Metric cant be None. Removed check.